TOOLS-1856 - use a memory pool in mongorestore

feliixx opened the pull request:
use a memory pool to avoid allocation for each
document to restore, see [TOOLS-1856](https://jira.mongodb.org/browse/TOOLS-1856) for details

tfogo commented:
Hi @feliixx, thank you for your PR. I'm so sorry it has taken this long for us to get back to you about this. It looks like your PR fell through the cracks and we never got round to reviewing it. But we're now going through our backlog of open PRs so hopefully we can get this merged.

Could you please sign the [MongoDB Contributor Agreement](https://www.mongodb.com/legal/contributor-agreement)? This will allow us to review and accept your PR.

feliixx commented:
Hi @tfogo, 

I've signed the contributor agreement, but I can't update this PR because I deleted my fork of mongo-tools a few years ago. I created a new PR and rebased the changes: https://github.com/mongodb/mongo-tools/pull/274

tfogo commented:
Thanks so much @feliixx, and thank you for doing the rebase too! I'll close this PR now since we have #274 open.
